### PR TITLE
Add support for stacked bar chart

### PR DESCRIPTION
**Pull request by mkrecek234**
Adding feature to support stacked bars.

Also additional default colour for longer datasets added.

### CODE REVIEW

File: src/Chart.php [hunk 1]
```diff
@@ -28,11 +28,15 @@ class Chart extends View
         ['rgba(75, 192, 192, 0.2)', 'rgba(75, 192, 192, 1)'],
         ['rgba(153, 102, 255, 0.2)', 'rgba(153, 102, 255, 1)'],
         ['rgba(255, 159, 64, 0.2)', 'rgba(255, 159, 64, 1)'],
+        ['rgba(20, 20, 20, 0.2)', 'rgba(20, 20, 20, 1)']
```
Comment (mvorisek) [inline src/Chart.php:31]: does this come from some design guide? can this can abstracted for unlimited colors?

File: src/Chart.php [hunk 2]
```diff
@@ -28,11 +28,15 @@ class Chart extends View
         ['rgba(75, 192, 192, 0.2)', 'rgba(75, 192, 192, 1)'],
         ['rgba(153, 102, 255, 0.2)', 'rgba(153, 102, 255, 1)'],
         ['rgba(255, 159, 64, 0.2)', 'rgba(255, 159, 64, 1)'],
+        ['rgba(20, 20, 20, 0.2)', 'rgba(20, 20, 20, 1)'],
```
Comment (mvorisek) [inline src/Chart.php:35]: does this come from some design guide? can this can abstracted for unlimited colors?

Comment (mkrecek234) [reply src/Chart.php:35]: Current Colour scheme looks nice and can well be used, I would not abstract or generalize it: devs should change if they prefer other colour coding

Comment (mvorisek) [reply src/Chart.php:35]: it seems the original colors come from https://www.chartjs.org/docs/3.9.1/#creating-a-chart example /wo any specific design guide/logic

**Comment by mvorisek** [comment]
please add a simple demo usage to `/demos/index.php`, this file is currently not tested by CI, but better to have a manual test than no test

**Comment by DarkSide666** [comment]
Added LineChart
Added BarChart->setHorizontal()
Improved demos

File: src/LineChart.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atk4\Chart;
+
+class LineChart extends Chart
```
Comment (mvorisek) [inline src/LineChart.php:7]: This complicates any extension in user projects... Not sure if different chart type that does not need additional modifications needs a custom class.

File: src/BarChart.php [hunk 4]
```diff
@@ -7,4 +7,15 @@
...
     public $type = 'bar';
+
+    /**
+     * Set this chart to be horizontal.
+     */
+    public function setHorizontal(): void
+    {
+        $this->type = 'horizontalBar';
+
+        // in chartjs 3.9.1 replace with
```
Comment (mvorisek) [inline src/BarChart.php:18]: better to upgrade to the latest lib now

File: src/LineChart.php (hunk 3 above)
Comment (DarkSide666) [reply src/LineChart.php:7]: There can be modifications in future is someone will be ready to contribute them :)
I think each chart type should have it's own class even if it is mostly empty. Also maybe some parts from generic Chart class should go to extended chart classes.

File: src/BarChart.php (hunk 4 above)
Comment (DarkSide666) [reply src/BarChart.php:18]: will do

File: src/LineChart.php (hunk 3 above)
Comment (mvorisek) [reply src/LineChart.php:7]: I am fine with this for now.

**Review by DarkSide666 — APPROVED** [review]

**Pull request merged**